src/main/java/com/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -1009,4 +1014,100 @@ public SymbolTable pop_passed_symbol_table()
+        // System readers don't handle symbol tables.
+    }
+
+    /**
+     * While reading an encoding directive, the reader allows itself to be controlled by the MacroCompiler during
+     * compilation of a macro. While this is happening, the reader should never attempt to read another encoding
+     * directive.
+     * @return true if the reader is not in the process of compiling a macro; false if it is.
+     */
+    private boolean macroCompilationNotInProgress() {
+        return encodingDirectiveReader == null || !encodingDirectiveReader.isMacroCompilationInProgress();
+    }
+
+    /**
+     * @return true if current value has a sequence of annotations that begins with `$ion_encoding`; otherwise, false.
+     */
+    boolean startsWithIonEncoding() {
+        // TODO also resolve symbol identifiers and compare against text that looks like $ion_encoding
+        return SystemSymbols_1_1.ION_ENCODING.getText().equals(_annotations[0].getText());
+    }
+
+    /**
+     * @return true if the reader is positioned on an encoding directive; otherwise, false.
+     */
+    private boolean isPositionedOnEncodingDirective() {
+        return _annotation_count > 0
+            && _value_type == IonType.SEXP
+            && !isNullValue()
+            && macroCompilationNotInProgress()
+            && startsWithIonEncoding();
+    }
+
+    /**
+     * Reads an encoding directive and installs any symbols and/or macros found within. Upon calling this method,
+     * the reader must be positioned on an s-expression annotated with `$ion_encoding`.
+     */
+    private void readEncodingDirective() {
+        if (encodingDirectiveReader == null) {
+            encodingDirectiveReader = new EncodingDirectiveReader(this);
+        }
+        encodingDirectiveReader.reset();
+        encodingDirectiveReader.readEncodingDirective();
+        List<String> newSymbols = encodingDirectiveReader.getNewSymbols();
+        if (encodingDirectiveReader.isSymbolTableAppend()) {
+            LocalSymbolTable current = ((LocalSymbolTable) getSymbolTable());
+            for (String appendedSymbol : newSymbols) {
+                current.putSymbol(appendedSymbol);
+            }
+        } else {
+            setSymbolTable(new LocalSymbolTable(
+                // TODO handle shared symbol table imports declared in the encoding directive
+                LocalSymbolTableImports.EMPTY,
+                newSymbols
+            ));
+        }
+        encodingContext = new EncodingContext(encodingDirectiveReader.getNewMacros());
+        macroEvaluator = new MacroEvaluator();
+        macroEvaluatorIonReader = new MacroEvaluatorAsIonReader(macroEvaluator);
+    }
+
+    /**
+     * Advances the reader, if necessary and possible, to the next value, reading any Ion 1.1+ encoding directives
+     * found along the way.
+     * @return true if the reader is positioned on a value; otherwise, false.
+     */
+    protected final boolean has_next_system_value() {
```

Comment:
This mimics the existing naming convention for the has_next methods. Previously, no `system`-level `has_next` was needed because the system-level reader did not interpret any system values. However, as in the binary reader, macro compilation and evaluation has to happen one layer below Ion 1.0 system table handling since Ion 1.1 macros can expand to system values.